app/src/main/java/helium314/keyboard/keyboard/KeyboardActionListener.java

```diff
@@ -97,6 +97,7 @@ public interface KeyboardActionListener {
      */
     boolean onHorizontalSpaceSwipe(int steps);
     boolean onVerticalSpaceSwipe(int steps);
+    boolean toggleNumpad(boolean withSliding, boolean forceReturnToAlpha);
```

Comment:
In general I would like to keep such interfaces small, and don't want this addition to be a starting point for adding more and more of these very specific actions.

But in this case I think it should be fine, if I understand it correctly:
`toggleNumpad` seems to be only needed for the _long-press symbols for numpad_ thing, because the sliding input can't be started when leaving the key, right?

Reply:
Sliding input is allowed for the symbols long-press. In that case, forceReturnToAlpha is true because otherwise the "previous" layout it returns to is 'symbols', which isn't really accurate since the long-press action was done in the alpha layout.

Reply:
If I need another param in the future, I may crunch them into an enumeration of "where to return" constants.